operators/src/processing/vector_join/equi_data_join.rs

```diff
@@ -0,0 +1,401 @@
+                    let feature_data = left
+                        .data(column_name)
+                        .expect("must exist")
+                        .get_unchecked(left_feature_idx);
+                    builder.push_data(column_name, feature_data)?;
+                }
+
+                // add right values
+                for column_name in right.column_names() {
+                    let feature_data = right
+                        .data(column_name)
+                        .expect("must exist")
+                        .get_unchecked(right_feature_idx);
+                    builder.push_data(column_name, feature_data)?;
+                }
+
+                builder.finish_row();
+            }
+        }
+
+        builder.build().map_err(Into::into)
+    }
+
+    /// Fix column by renaming column with duplicate keys
+    fn fix_columns(
+        &self,
+        collection: &DataCollection,
+        existing_column_names: &HashSet<&String>,
+    ) -> Result<DataCollection> {
+        let mut renamings: Vec<(&str, String)> = Vec::new();
+
+        for column_name in collection.column_names() {
+            while existing_column_names.contains(column_name) {
+                let new_column_name = format!("{}{}", self.right_column_prefix, column_name);
+                renamings.push((column_name, new_column_name));
+            }
+        }
+
+        if renamings.is_empty() {
+            Ok(collection.clone())
+        } else {
+            collection.rename_columns(&renamings).map_err(Into::into)
+        }
+    }
+}
+
+impl<G> VectorQueryProcessor for EquiLeftJoinProcessor<G>
+where
+    G: Geometry + ArrowTyped + Sync + Send + 'static,
+    for<'g> FeatureCollection<G>: IntoGeometryIterator<'g>,
+    for<'g> <FeatureCollection<G> as IntoGeometryIterator<'g>>::GeometryType: Into<G>,
+    FeatureCollectionRowBuilder<G>: GeoFeatureCollectionRowBuilder<G>,
+{
+    type VectorType = FeatureCollection<G>;
+
+    fn vector_query(
+        &self,
+        query: QueryRectangle,
+        ctx: QueryContext,
+    ) -> BoxStream<Result<Self::VectorType>> {
+        let result_stream =
+            self.left_processor
+                .query(query, ctx)
+                .flat_map(move |left_collection| {
+                    // This implementation is a nested-loop join
+
+                    let left_collection = match left_collection {
+                        Ok(collection) => collection,
+                        Err(e) => return stream::once(async { Err(e) }).boxed(),
+                    };
+
+                    let data_query = self.right_processor.query(query, ctx);
+
+                    data_query
+                        .map(move |right_collection| {
+                            self.join(&left_collection, &right_collection?)
+                        })
+                        .boxed()
+                });
+
+        FeatureCollectionChunkMerger::new(result_stream.fuse(), ctx.chunk_byte_size).boxed()
```

Comment:
You do not only have to make sure the chunks are at least chunk_byte_size large but also that they are not (much) larger. The join result may be much larger than the individual collections.

Reply:
needed to introduce Iterator and random geo access

7d53833

Reply:
why exactly was the random access necessary? Couldn't you store an iterator and the current feature when a chunk is full? The random access on features looks expensive to me, but I could be wrong.

Reply:
It is suboptimal, but I can't store the collection and an iterator that references the collection in the same (iterator) struct. If you have an idea on how to accomplish this, we can try it out.

Reply:
I created a moonshot task https://trello.com/c/WKrE6EyY